src/routers/documents/documents.py

```diff
@@ -0,0 +1,75 @@
+from typing import Annotated
+from fastapi import APIRouter, Depends, HTTPException, Request, Response, UploadFile
+from src.dependecies import get_db
+from sqlalchemy.orm import Session
+from src.services.document_handler import DocumentHandler
+from starlette import status
+from src.routers.documents.schemas import Document
+from src.services.auth_utils import check_privilege
+
+documents_router = APIRouter()
+
+@documents_router.get("/document/{document_id}")
+async def get_document(request: Request,
+                       document_id: int,
+                       db: Annotated[Session, Depends(get_db)]):
+    # check if document exists
+    project_id = DocumentHandler.check_document_exists(document_id=document_id,
+                                                       db=db)
+    # check if caller is authorized to get it
+    owned = request.state.owned
+    participating = request.state.participating
+    check_privilege(project_id=project_id,
+                    owned_projects=owned,
+                    participating_projects=participating)
+    # call method
+    name, content_type, contents = DocumentHandler.download_document(document_id=document_id, db=db)
+    return Response(

```

Comment:
Initially I planned to use ```FileResponse``` here. However, when I got to the actual implementation, I realized that the way it takes the file to return is via a ```path``` parameter. Looking at the docs (both FastAPI and Stralette) it is clear that it expects a ```str``` containing the path to the file to return (like ```~/files/file1.txt```) *on disk*. Since I'm not storing the files in the same place where my API service will be running, I didn't want to store anything on disk, but to work with files *in memory*, and after doing extensive research on how to use ```FileResponse``` with file-like objects, I didn't find any solution, but a mitigating measure to use the ```Response``` class and configure the header and contents in such a way that it causes a file download on client-side.